diff --git a/config/arm/meson.build b/config/arm/meson.build
index 5213434ca4..b4410585be 100644
--- a/config/arm/meson.build
+++ b/config/arm/meson.build
@@ -574,7 +574,7 @@ else
         endif
     else
         # cross build
-        soc = meson.get_cross_property('platform', '')
+        soc = meson.get_cross_property('platform', 'generic')
         if soc == ''
             error('Arm SoC must be specified in the cross file.')
         endif
diff --git a/lib/meson.build b/lib/meson.build
index 0812ce6026..2d51ed3d06 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -73,7 +73,6 @@ optional_libs = [
         'gpudev',
         'gro',
         'gso',
-        'kni',
         'jobstats',
         'latencystats',
         'metrics',
@@ -88,15 +87,11 @@ optional_libs = [
 
 dpdk_libs_deprecated += [
         'flow_classify',
-        'kni',
 ]
 
 disabled_libs = []
 opt_disabled_libs = run_command(list_dir_globs, get_option('disable_libs'),
         check: true).stdout().split()
-if dpdk_conf.get('RTE_IOVA_IN_MBUF') == 0
-    opt_disabled_libs += ['kni']
-endif
 foreach l:opt_disabled_libs
     if not optional_libs.contains(l)
         warning('Cannot disable mandatory library "@0@"'.format(l))
diff --git a/meson_options.txt b/meson_options.txt
index 82c8297065..b51689a343 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -10,7 +9,7 @@ option('disable_apps', type: 'string', value: '', description:
        'Comma-separated list of apps to explicitly disable.')
 option('disable_drivers', type: 'string', value: '', description:
        'Comma-separated list of drivers to explicitly disable.')
-option('disable_libs', type: 'string', value: 'flow_classify,kni', description:
+option('disable_libs', type: 'string', value: 'flow_classify', description:
        'Comma-separated list of libraries to explicitly disable. [NOTE: not all libs can be disabled]')
 option('drivers_install_subdir', type: 'string', value: 'dpdk/pmds-<VERSION>', description:
        'Subdirectory of libdir where to install PMDs. Defaults to using a versioned subdirectory.')
